pkg/workloads/cortex/lib/storage/s3.py

```diff
@@ -250,6 +250,10 @@ def get_json_external(self, s3_path, num_retries=0, retry_delay_sec=2):
             return None
         return json.loads(obj.decode("utf-8"))
 
-    def list_objects(self, s3_path):
-        bucket, prefix = self.deconstruct_s3_path(s3_path)
-        return self.s3.list_objects(Bucket=bucket, Prefix=prefix)
+    def list_objects(self, path):
```

Comment:
Do we need to handle continuation tokens?

Reply:
EDIT: I remember dealing with this once, I think we can use `search()` in this file for `list_objects()` (which correctly handles continuation tokens). The only issue is that assumes the internal bucket, so maybe you can generalize that?

Reply:
EDIT2: maybe we can instantiate a new `storage` client in the appropriate bucket in `api.py` and then we wouldn't have to modify `search()`